docs/config-overview.md

```diff
@@ -0,0 +1,31 @@
+# Terramate Configuration Overview
+
+There is a series of different configurations that can be done
+on Terramate, ranging from avoiding duplication by leveraging
+powerful code generation to flexible orchestration by allowing
+control of stacks order of execution.
+
+In order to do so, Terramate works with configuration files that
+have the suffixes:
+
+* `tm`
+* `tm.hcl`
+
+Terramate files can be found in any directory of a Terramate project and
+all the files in a single directory will be handled as the concatenation
+of all of them in a single file, forming a single **configuration**.
+
+A Terramate project is essentially any collection of Terraform code
```

Comment:
```
A Terramate project is essentially a collection of normal Terraform code and 
terramate files organized into stacks.
```
?

Reply:
I went only with "Terraform code"...I think it is clear enough that Terraform Code should be normal, abnormal Terraform code is not Terraform code, it would be HCL.

Reply:
If it's clear enough it's normal TF code, then the next phrase could be dropped... `The stacks can also reference local modules inside the project or remote ones.`.
I proposed adding "normal TF code ..." to avoid having to explain he/she can use modules, etc.

Reply:
I also think that saying "The stacks can also reference local modules ..." is confusing if you're talking about terramate stacks or terraform stuff, because the previous phrase starts by defining a Terramate project.

Reply:
Yeah I just removed that, terraform code can always use modules, so also redundant/confusing.